### ISSUE TITLE
Switch: java.lang.ClassCastException: android.graphics.drawable.ColorDrawable cannot be cast to android.graphics.drawable.RippleDrawable	

### ISSUE THREAD

**Issue by tapz** [facebook/react-native#33020]
### Description

If thumbColor and style attributes are both set to a Switch in Android the following exception is thrown:

```
Exception in native call	
java.lang.ClassCastException: android.graphics.drawable.ColorDrawable cannot be cast to android.graphics.drawable.RippleDrawable	
	at com.facebook.react.views.switchview.ReactSwitch.setThumbColor(ReactSwitch.java:68)	
	at com.facebook.react.views.switchview.ReactSwitchManager.setThumbColor(ReactSwitchManager.java:153)	
	at com.facebook.react.views.switchview.ReactSwitchManager.setThumbTintColor(ReactSwitchManager.java:147)	
	at com.facebook.react.views.switchview.ReactSwitchManager.setThumbTintColor(ReactSwitchManager.java:35)	
	at com.facebook.react.viewmanagers.AndroidSwitchManagerDelegate.setProperty(AndroidSwitchManagerDelegate.java:48)	
	at com.facebook.react.uimanager.ViewManagerPropertyUpdater.updateProps(ViewManagerPropertyUpdater.java:46)	
	at com.facebook.react.uimanager.ViewManager.updateProperties(ViewManager.java:45)	
	at com.facebook.react.uimanager.NativeViewHierarchyManager.updateProperties(NativeViewHierarchyManager.java:142)	
	at com.facebook.react.uimanager.UIViewOperationQueue$UpdatePropertiesOperation.execute(UIViewOperationQueue.java:93)	
	at com.facebook.react.uimanager.UIViewOperationQueue$1.run(UIViewOperationQueue.java:915)	
	at com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches(UIViewOperationQueue.java:1026)	
	at com.facebook.react.uimanager.UIViewOperationQueue.access$2600(UIViewOperationQueue.java:47)	
	at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:1086)	
	at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java:29)	
	at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:175)	
	at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java:85)	
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1008)	
	at android.view.Choreographer.doCallbacks(Choreographer.java:809)	
	at android.view.Choreographer.doFrame(Choreographer.java:740)	
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:995)	
	at android.os.Handler.handleCallback(Handler.java:938)	
	at android.os.Handler.dispatchMessage(Handler.java:99)	
	at android.os.Looper.loop(Looper.java:246)	
	at android.app.ActivityThread.main(ActivityThread.java:8633)	
	at java.lang.reflect.Method.invoke(Native Method)	
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)	
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)	
```

### Version

0.67.2

### Output of `npx react-native info`

System:
    OS: macOS 12.1
    CPU: (8) x64 Intel(R) Core(TM) i7-1068NG7 CPU @ 2.30GHz
    Memory: 377.20 MB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.18.1 - /usr/local/opt/node@14/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 6.14.15 - /usr/local/opt/node@14/bin/npm
    Watchman: 2021.11.15.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.2 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.2, iOS 15.2, macOS 12.1, tvOS 15.2, watchOS 8.3
    Android SDK:
      API Levels: 25, 26, 27, 28, 29, 30, 31
      Build Tools: 30.0.3
      Android NDK: Not Found
  IDEs:
    Android Studio: 2020.3 AI-203.7717.56.2031.7935034
    Xcode: 13.2.1/13C100 - /usr/bin/xcodebuild
  Languages:
    Java: 14.0.2 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.67.2 => 0.67.2 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

Create a Switch.

### Snack, code example, screenshot, or link to a repository

```
<Switch
  style={styles.checkMark}
  thumbColor='red'
  trackColor='blue'
  onValueChange={onAvailable}
  value={available} />

const styles = StyleSheet.create({
  checkMark: {
    backgroundColor: 'red',
    width: 80
  }
});
```

**Comment by fabriziobertoglio1987** [comment]
```
android.graphics.drawable.ColorDrawable cannot be cast to android.graphics.drawable.RippleDrawable
```
https://github.com/facebook/react-native/blob/da420c99acea69e7c6a08795385b8a16df0b2853/ReactAndroid/src/main/java/com/facebook/react/views/switchview/ReactSwitch.java#L63-L68

I think was fixed with https://github.com/fabriziobertoglio1987/react-native-notes/commit/456cf3db14c443c483d63aa97c88b45ffd25799b

**Comment by cortinico** [comment]
Agree. This was fixed by:
- https://github.com/facebook/react-native/pull/32468

It will be included inside RN 0.68.x (it's already included in the RC0 & 1 if you want to give it a try @tapz)
